test/utils/describeConformanceUnstyled.tsx

```diff
@@ -249,6 +249,37 @@ function testComponentsPropsProp(
   });
 }
 
+function testComponentsPropsCallbacks(
+  element: React.ReactElement,
+  getOptions: () => UnstyledConformanceOptions,
+) {
+  const { render, slots } = getOptions();
+
+  if (!render) {
+    throwMissingPropError('render');
+  }
+
+  if (!slots) {
+    throwMissingPropError('slots');
+  }
+
+  forEachSlot(slots, (slotName, slotOptions) => {
+    it(`sets custom properties on ${capitalize(
+      slotName,
+    )} slot's element with a callback function`, () => {
+      const componentsProps = {
+        [slotName]: () => ({
+          'data-testid': 'custom',
```

Comment:
Hmm, I would expect that you would set a prop on the component itself, for example: `data-testid="parent"` and then use that in the callback for example for setting data-testid on the slot in form of: 'data-testid': `${props['data-testid']}-${slotName}`. This way we will check that both the callback works and the props are propagated.

Reply:
Good idea, thanks!